cmd/m3/cluster-sc-sg-add.go

```diff
@@ -45,17 +45,25 @@ func addStorageGroup(ctx *cli.Context) error {
 		name = &nameVal
 	}
 
+	appCtx, err := cluster.NewContext("")
+	if err != nil {
+		return err
+	}
+
 	// create a new storage group in the DB
-	storageGroupResult := <-cluster.AddStorageGroup(name)
+	storageGroupResult := <-cluster.AddStorageGroup(appCtx, name)
 	if storageGroupResult.Error != nil {
 		fmt.Println(storageGroupResult.Error)
+		appCtx.Rollback()
```

Comment:
[1] here there will be nothing to rollback if addStorageGroup already rolled back

Reply:
It's harmless if things get rolled back twice the `ctx.Rollback()` will handle being tried to rollback a non existing transaction (as it gets closed on the previous rollback)

Reply:
also if the internal call doesn't rollback the invoking function may decide to rollback, so it's fine to have both